### ISSUE TITLE
Turn off 'Rewind' function by default otherwise lots of games won't boot

### ISSUE THREAD

**Issue by ghost** [libretro/mame2003-plus-libretro#175]
So after over 110 comments from this thread https://github.com/libretro/mame2003-plus-libretro/issues/163 I finally found out that an option called 'Rewind' was preventing lots of games from booting. Maybe MAME 2003+ should disable this option by default otherwise lots of games won't boot ?

The thing is, I never even enabled this option so I don't know if it's on by default in RetroArch ?

**Comment by dankcushions** [comment]
rewind is turned off by default in retropie, but i'm guessing at some point you deleted your retropie configs, and now retroarch defaults were enabled, where it is on by default.

i wonder why it's causing the issues, though - some games in mame2003 do not support save states, which is required for rewind. are these the games causing it to crash? i think once we have a scenario for the crash, we should report it to retroarch, if that's indeed where the issue is.

**Comment by ZappaUtopia** [comment]
I checked my savestate options and they were all turned off. I know for a fact that 3 of the games I listed in the other thread previously worked fine a couple of weeks ago, so if the Rewind function was On for me then, the games still worked. One of the commits since then must have made these few games exit if Rewind is on ? I have no idea.

**Comment by dankcushions** [comment]
> I checked my savestate options and they were all turned of

save state options?

> I know for a fact that 3 of the games I listed in the other thread previously worked fine a couple of weeks ago, so if the Rewind function was On for me then, the games still worked

someone could test an older version (maybe testing regular mame2003 would work?), with rewind on in these games and see what happens?

with all the debugging going i am still guessing at my previous theory about retroarch defaults sneaking in to your install.

**Comment by dankcushions** [comment]
rewind enable i think is in another menu (Settings —> Frame Throttle —> Rewind, maybe?), so save state options won't matter.

**Comment by ZappaUtopia** [comment]
I suppose it doesn't really matter now anyway. If one of you guys could make Mame2003+ turn off the 'Rewind' feature when running then no-one will encounter certain games not running even if they have the Rewind option set on different emulators.

**Comment by dankcushions** [comment]
a core cannot enforce front end options, but like i said i think a crash would be a retroarch bug that we could report, if that's what's going on.

**Comment by ZappaUtopia** [comment]
Shall I close this then as it was already reported for 2003 or leave it open as an issue with 2003+ as well ?

**Comment by grant2258** [comment]
i would leave this open until we get a conclusion @ZappaUtopia

**Comment by KMFDManic** [comment]
@ZappaUtopia This may be something relative to 1.7.2 RetroArch, as I have not had this issue, to my recollection in 1.7.1 and earlier. Games would simply be "slower" til I disabled Rewind/Frame Throttle:)

**Comment by grant2258** [comment]
@arcadez this issue looks like the one we had with the neogeo games that would work check the video at 24:47

**Comment by arcadez** [comment]
@grant2258 Metal Slug 5 etc etc were reported as not working in another project using the
exact same code, plus when i asked @ZappaUtopia what the problems were exactly the game
didn't even get as far as the loading screen it just failed and dropped back to the main romlist.

So i kinda think rewind being enabled wasn't the issue maybe more a case of the Neogeo
core from MAME2003 not being correct for those games to be able to function.

Regards.

**Comment by arcadez** [comment]
im on the case now with regards to adding it back in, i will however be updating the neogeo bios
just to rule out leaving the older bios handling in the driver being the cause

**Comment by arcadez** [comment]
Alright please test mslug5 firstly create a new dat trackdown the correact rom using it
and remember you'll need to update neogeo bios also, if it doesn't work after this it
never will in this core

**Comment by grant2258** [comment]
compiled now will created the dat and report back mr arcadez.

**Comment by arcadez** [comment]
cool if you need help tracking down either the rom or neogeo bios drop me a line

**Comment by grant2258** [comment]
i have the roms just trying to track down the bios now @arcadez

**Comment by grant2258** [comment]
@arcadez i found them mate there is a v3.3 available now as well just for your info you probably know that already time to test!

**Comment by grant2258** [comment]
im getting a failed to load content when trying this other neogeo are loading fine though

**Comment by grant2258** [comment]
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Metal Slug 5", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Metal Slug 5.zip", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5.zip", O_RDONLY) = 3
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5.zip", O_RDONLY) = 3
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Metal Slug 5", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Metal Slug 5.zip", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5.zip", O_RDONLY) = 3
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5.zip", O_RDONLY) = 3
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Metal Slug 5", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Metal Slug 5.zip", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5.zip", O_RDONLY) = 3
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Neo-Geo", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Neo-Geo.zip", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//neogeo", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//neogeo.zip", O_RDONLY) = 3
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//neogeo.zip", O_RDONLY) = 3
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Metal Slug 5", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//Metal Slug 5.zip", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5.zip", O_RDONLY) = 3
[pid  1568] open("/media/grant/empty/Download/Emulation/roms/arcade/mame2003-plus//mslug5.zip", O_RDONLY) = 3
[libretro ERROR] [MAME 2003+] Error in RomModule definition: 268-m1.bin out of memory region space
[libretro ERROR] [MAME 2003+] readroms failed

**Comment by grant2258** [comment]
might be something we have done will look into it more my arcadez

**Comment by grant2258** [comment]
ok mr arcadez we are good to go and it works it was the // cleanup that broke it

**Comment by markwkidd** [comment]
Once upon a time I created a new driver flag for games that required special initialization like using F2 etc. Now that we have the nvram boostrap system that flag is moot.

Therefore I re-purposed it into a flag to indicate a problem with the libretro serialization support. On a game-by-game basis this might be that MAME 0.78 doesn't support saves, or that there is an issue with the libretro implementation. Basically, the usual situation.

Now if we flag the affected drivers with `GAME_DOESNT_SERIALIZE`, then `retro_serialize` will return false and there will be no crash. In effect it will disable Rewind, but just for that title. [Here's how I implemented the new flag for Sexy Pariodius as a proof of concept](https://github.com/libretro/mame2003-plus-libretro/blob/64c6c4af24b43bab94f7928f6591b82c48c3470e/src/drivers/konamigx.c#L3453)
